### PR TITLE
Expand Go section

### PR DESCRIPTION
**Pull request by rhcarvalho**

### CODE REVIEW

File: tech/languages/go/go-installation.md [hunk 1]
```diff
@@ -3,31 +3,111 @@ title: Go
...
+$ sudo dnf install golang
+```
+
+The `go` and `gofmt` binaries will become available on the system.
+
+Go code lives in [a workspace](https://golang.org/doc/code.html#Workspaces) which is defined by the `GOPATH` environment variable. A common choice among developers, and the [default value of `GOPATH` starting from the Go 1.8 release](https://tip.golang.org/doc/code.html#GOPATH), is to use `$HOME/go`:
+
+```console
+$ mkdir -p $HOME/go
+$ echo 'export GOPATH=$HOME/go' >> $HOME/.bashrc
```
Comment (pvalena) [inline tech/languages/go/go-installation.md]: Hello, first, let me thank you for your PR. Could you please use the `~/.bash_profile` here? Putting the path into `~/.bashrc` seems unfitting. Please also see my previous conversation here:

https://github.com/developer-portal/content/pull/169#pullrequestreview-10301333

File: tech/languages/go/go-installation.md [hunk 2]
```diff
@@ -3,31 +3,111 @@ title: Go
...
+
+```console
+$ go env GOPATH
+/home/user/go
 ```
 
-`go` and `gofmt` binaries will become available on the system.
+Where 'user' will be your user name.
+
+## Writing programs
```
Comment (pvalena) [inline tech/languages/go/go-installation.md]: I think this is worth standalone subsection.

File: tech/languages/go/go-packages.md [hunk 3]
```diff
@@ -8,16 +8,16 @@ order: 2
...
 ```
 $ go get github.com/gorilla/context
 ```
 
-If the requested package comes with an executable binary, you should find it in `$GOPATH/bin` directory. To use it, you need to add this path to your `$PATH`:
+If the requested package comes with an executable binary, you will find it in `$GOPATH/bin`. If you have not added that to your `PATH` environment variable yet, you can do so with these commands:
 
 ```
-$ echo 'export PATH=$GOPATH/bin:$PATH' >> $HOME/.bashrc
+$ echo 'export PATH=$PATH:$GOPATH/bin' >> $HOME/.bashrc
```
Comment (pvalena) [inline tech/languages/go/go-packages.md]: Same here as above. Please see https://github.com/developer-portal/content/pull/169#pullrequestreview-10301333

**Comment by rhcarvalho** [comment]
Sorry for taking a long time to answer, I've been lacking spare time to look at this, hope to do the proposed changes some time soon.

**Comment by phracek** [comment]
@rhcarvalho Can you please update it, so we can merge it?

File: tech/languages/go/go-installation.md (hunk 1 above)
Comment (rhcarvalho) [inline tech/languages/go/go-installation.md]: Could you clarify why is PATH in `~/.bashrc` unfitting? That's how I have it in my system, and `~/.bash_profile` sources `~/.bashrc`. If there's a reason not to set/export PATH in `~/.bashrc` I'd like to learn.

I remember I explicitly looked around to keep some level of consistency with other pages. Both Rust and Rails contain very similar instructions.

If with the link to another comment you mean `export` is not needed, I disagree. We want new processes to inherit the GOPATH variable.

Comment (rhcarvalho) [inline tech/languages/go/go-installation.md]: > I remember I explicitly looked around to keep some level of consistency with other pages. Both Rust and Rails contain very similar instructions.

And one of the Go pages itself:
https://developer.fedoraproject.org/tech/languages/go/go-packages.html

```
$ echo 'export PATH=$GOPATH/bin:$PATH' >> $HOME/.bashrc
$ source $HOME/.bashrc
```

File: tech/languages/go/go-installation.md (hunk 2 above)
Comment (rhcarvalho) [inline tech/languages/go/go-installation.md]: Done.

**Comment by rhcarvalho** [comment]
@phracek updated.

**Comment by rhcarvalho** [comment]
Hi, anything else here?

File: tech/languages/go/go-installation.md (hunk 1 above)
Comment (pvalena) [inline tech/languages/go/go-installation.md]: Sorry about not responding. You're right, we need to keep it consistent. I will create an issue and we will fix all the instructions.

**Comment by pvalena** [comment]
@phracek thanks!

You can check your changes here:

http://developer.fedorainfracloud.org/tech/languages/go/go-installation.html
http://developer.fedorainfracloud.org/tech/languages/go/go-programs.html
http://developer.fedorainfracloud.org/tech/languages/go/go-packages.html

**Pull request merged**